spec/models/container_label_tag_mapping_spec.rb

```diff
@@ -184,37 +184,234 @@ def map_to_tags(mapper, model_name, labels_kv)
+        end
+
+        context "when mappings are targeted to different categories" do
+          it "adds tags to resource" do
+            expect(vm.tags).to be_empty
+
+            subject
+
+            expected_tags = %w[location/brno environment/accounting].map { |x| Tag.lookup_by_classification_name(x) }
+            expect(vm.reload.tags).to match_array(expected_tags)
+          end
+        end
+
+        context "when mappings are targeted to same categories" do
+          let(:externals_labels) do # provider label
+            [{:name => 'Name', :value => 'Accounting'},
+             {:name => 'Location', :value => 'Production'}]
+          end
+
+          before do
+            FactoryBot.create(:container_label_tag_mapping, :all_entities, :label_name => "Location", :tag => single_value_category.tag)
+          end
+
+          it "only applies one of the label values" do
+            expect(vm.tags).to be_empty
+
+            subject
+
+            expected_tags = %w[environment/accounting environment/production].map { |x| Tag.lookup_by_classification_name(x) }
+            expect(vm.reload.tags.count).to eq(1)
+
+            # The order in which the labels are mapped in not deterministic. Therefore, only one tag int he category should be applied
+            # and it can be either one
+            expect(vm.reload.tags.first).to eq(expected_tags.first).or(be == expected_tags.second)
+          end
         end
       end
     end
-  end
 
-  context "with 2 mappings to the same category (implied multi-value)" do
-    it "should be valid" do
+    context "with mapping with multi-value category" do
+      let!(:multi_value_category_2) { FactoryBot.create(:classification_cost_center_with_tags) }
 
-    end
+      let(:externals_labels) do # provider label
+        [{:name => 'Name', :value => 'Accounting'}]
+      end
 
-    it "should add new tag during mapping" do
+      before do
+        FactoryBot.create(:container_label_tag_mapping, :all_entities, :label_name => "Name", :tag => multi_value_category.tag)
+      end
 
-    end
+      it "adds a tag to the resource with the value of the external label" do
+        expect(vm.tags).to be_empty
+
+        subject
+
+        expected_tags = [Tag.lookup_by_classification_name('department/accounting')]
+        expect(vm.reload.tags).to eq(expected_tags)
+      end
+
+      it "replaces pre-assigned tag with same category as target category from mapping" do
+        expect(vm.tags).to be_empty
+
+        existing_tag_on_vm = Tag.lookup_by_classification_name('department/hr')
+        vm.tag_add(existing_tag_on_vm.name, :ns => '')
+        expect(vm.reload.tags).to eq([existing_tag_on_vm])
+
+        subject
+
+        expect(vm.reload.tags).to match_array([Tag.lookup_by_classification_name('department/accounting')])
+      end
+
+      it "add tag with same category as target category from mapping" do
```

Comment:
Should be  "add tag with a category that is different from the target category from mapping"